src/megachatapi_impl.cpp

```diff
@@ -1821,6 +1821,12 @@ int MegaChatApiImpl::init(const char *sid)
         localLogout();
     }
 
+    if (!waitForFetchnodesToConnect && state != karere::Client::kInitHasOfflineSession)
+    {
+        localLogout();
+        state = karere::Client::kInitErrGeneric;
```

Comment:
Better to use a `setState(state)` here

Reply:
Actually, the best would be to manage the error internally by `karere::Client`. I've pushed a commit to do so. It does not affect the logic from the app's point of view :)